src/datapoints-graph/charts/charts.component.ts

```diff
@@ -143,6 +162,242 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
+        if (!originalFormatter) {
+          originalFormatter = options.tooltip[0].formatter;
+        }
+        const updatedOptions: Partial<SeriesOption> = {
+          tooltip: options.tooltip,
+        };
+        updatedOptions.tooltip[0].formatter = (tooltipParams) => {
+          const XAxisValue: string = tooltipParams[0].data[0];
+          const YAxisReadings: string[] = [];
+          const allSeries = this.echartsInstance.getOption()
+            .series as CustomSeriesOptions[];
+
+          const allDataPointSeries = allSeries.filter(
+            (series) =>
+              series.typeOfSeries !== 'alarm' && series.typeOfSeries !== 'event'
+          );
+
+          allDataPointSeries.forEach((series: any) => {
+            let value: string;
+            if (series.id.endsWith('/min')) {
+              const minValue = this.findValueForExactOrEarlierTimestamp(
+                series.data,
+                XAxisValue
+              );
+              if (!minValue) {
+                return;
+              }
+              const maxSeries = allDataPointSeries.find(
+                (s) => s.id === series.id.replace('/min', '/max')
+              );
+              const maxValue = this.findValueForExactOrEarlierTimestamp(
+                maxSeries.data as SeriesValue[],
+                XAxisValue
+              );
+              value =
+                `${minValue[1]} — ${maxValue[1]}` +
+                (series.datapointUnit ? ` ${series.datapointUnit}` : '') +
+                `<div style="font-size: 11px">${this.datePipe.transform(
+                  minValue[0]
+                )}</div>`;
+            } else if (series.id.endsWith('/max')) {
+              // do nothing, value is handled  in 'min' case
+              return;
+            } else {
+              const seriesValue = this.findValueForExactOrEarlierTimestamp(
+                series.data,
+                XAxisValue
+              );
+              if (!seriesValue) {
+                return;
+              }
+              value =
+                seriesValue[1]?.toString() +
+                (series.datapointUnit ? ` ${series.datapointUnit}` : '') +
+                `<div style="font-size: 11px">${this.datePipe.transform(
+                  seriesValue[0]
+                )}</div>`;
+            }
+
+            YAxisReadings.push(
+              `<span style='display: inline-block; background-color: ${series.itemStyle.color} ; height: 12px; width: 12px; border-radius: 50%; margin-right: 4px;'></span>` + // color circle
+                `<strong>${series.datapointLabel}: </strong>` + // name
+                value // single value or min-max range
+            );
+          });
+
+          const event = this.events.find(
+            (e) => (e.type = params.data.eventType)
+          );
+          const alarm = this.alarms.find(
+            (a) => (a.type = params.data.alarmType)
+          );
+
+          let value: string;
+          if (event) {
```

Comment:
Has @carlosceia seen this tooltip? looks like a draft view (events data). data should be grouped in some way (not to repeat 'Event...' and date needs formatting.